src/geometry/java/us/ihmc/euclid/geometry/tools/EuclidGeometryTools.java

```diff
@@ -4040,6 +4043,73 @@ public static boolean intersectionBetweenLine2DAndLineSegment2D(Point2DReadOnly
                                                        lineSegmentEnd.getY(),
                                                        intersectionToPack);
    }
+   
+   
+   /**
+    * Flexible implementation for computing the intersection between a ray/line/line-segment and
+    * point shape 3D.
+    * <p>
+    * <ul>
+    *
+    *
+    * @param pointX                           the x-coordinate of the point
+    * @param pointY                           the y-coordinate of the point
+    * @param pointZ                           the z-coordinate of the point
+    * 
+    * 
+    * @param lineStartX                       the x-coordinate of a point located on the
+    *                                         line/line-segment/ray.                                        
+    * @param lineStartY                       the y-coordinate of a point located on the
+    *                                         line/line-segment/ray.
+    * @param lineStartZ                       the z-coordinate of a point located on the
+    *                                         line/line-segment/ray.                                                                                
+    
+
+    * @param lineDirectionX                         the x-coordinate of a point located on the
+    *                                         line/line-segment/ray                                        
+    * @param lineDirectionY                         the y-coordinate of a point located on the
+    *                                         line/line-segment/ray
+    * @param lineDirectionZ                         the z-coordinate of a point located on the
+    *                                         line/line-segment/ray    
+    *                                       
+    * @param firstIntersectionToPack               the coordinate of the intersection. Can be {@code null}.
+    *                                         Modified.
+    
+    */
+   public static int intersectionBetweenLine3DAndPointShape3DImpl(double pointX,
+                                                                 double pointY,
+                                                                 double pointZ,
+                                                                 double lineStartX,
+                                                                 double lineStartY,
+                                                                 double lineStartZ,
+                                                                 double lineDirectionX,
+                                                                 double lineDirectionY,
+                                                                 double lineDirectionZ,
```

Comment:
This is only testing for the distance between line and point, not sure that's worth implementing a new method for it, we could simply rely on the distance method directly and implement the intersection in `PointShape3DReadOnly` directly.